piccolo/apps/schema/commands/generate.py

```diff
@@ -307,6 +308,12 @@ def __add__(self, value: OutputSchema) -> OutputSchema:
     "uuid": UUID,
 }
 
+# Re-map for Cockroach compatibility.
+if engine.engine_type == 'cockroach':
+    COLUMN_TYPE_MAP["integer"] = BigInt
+    COLUMN_TYPE_MAP["json"] = JSONB
+    COLUMN_TYPE_MAP["smallint"] = BigInt
```

Comment:
Should `smallint` be mapped to `BigInt`?

https://www.cockroachlabs.com/docs/stable/int.html#names-and-aliases

I'm not sure about modifying the dictionary - even though it works, maybe we can figure out a different approach.

Something like:

```python
COLUMN_TYPE_MAP = {
    'postgres': {...},
    'cockroach': {...}
}
```

Or:

```python
COCKROACH_COLUMN_TYPE_MAP = dict(
    COLUMN_TYPE_MAP,
    integer=BigInt,
    json=JSONB,
    smallint=BigInt
)
```

We would then have to pass around the engine_type to each function.

Reply:
Great ideas!

Reply:
Added. Gonna see if I can get SmallInt working.